Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift

```diff
@@ -85,6 +97,69 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
                     try statement.bind(bindings)
                     try statement.step()
                 }
+                
+                // Update search indices
+                try self.ftsLock.withLock {
```

Comment:
how is this lock used?

Reply:
I explain it in the code comment below but this prevents concurrency errors with `BEGIN TRANSACTION` since there is only one db connection and each connection can have just one transaction. e.g., I ran into this problem in the test for `refreshCollections`.